src/serializers/LogSerializer.ts

```diff
@@ -63,22 +71,48 @@ export class LogSerializer implements ISerializer {
     let currentBody = createBody();
 
     const currentMetricsInBody = (): number => currentBody._aws.CloudWatchMetrics[0].Metrics.length;
-    const shouldSerialize = (): boolean => currentMetricsInBody() === Constants.MAX_METRICS_PER_EVENT;
+    const hasMaxMetrics = (): boolean => currentMetricsInBody() === Constants.MAX_METRICS_PER_EVENT;
 
     // converts the body to JSON and pushes it into the batches
     const serializeCurrentBody = (): void => {
       eventBatches.push(JSON.stringify(currentBody));
       currentBody = createBody();
     };
 
-    for (const [key, metric] of context.metrics) {
-      // if there is only one metric value, unwrap it to make querying easier
-      const metricValue = metric.values.length === 1 ? metric.values[0] : metric.values;
-      currentBody[key] = metricValue;
-      currentBody._aws.CloudWatchMetrics[0].Metrics.push({ Name: key, Unit: metric.unit });
+    const remainingMetrics = MaxHeap.heapify<MetricProgress>(
```

Comment:
Could you please provide more explanation on why `MaxHeap` is used here ? Can we not use the existing `for` loop like we do in the python library https://github.com/awslabs/aws-embedded-metrics-python/blob/master/aws_embedded_metrics/serializers/log_serializer.py#L69

Reply:
The reason is I want to minimize the number of requests. For example, if there are 101 metrics, with the first 100 metrics each has 1 data point and the last metric has 101 data points. If you only use for loop, you will end up creating 3 requests:
1. 100 metrics
2. 100 data points of 101st metric
3. Last data point of 101st metric

However if you use max heap, you will end up with 2 requests:
1. 100 data points of 101st metric with the other 99 metrics
2. Last data point of 101st metric with the last metric